protocol/statetracker/event_tracker.go

```diff
@@ -0,0 +1,114 @@
+package statetracker
+
+import (
+	"context"
+	"sync"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/lavanet/lava/protocol/rpcprovider/reliabilitymanager"
+	"github.com/lavanet/lava/protocol/rpcprovider/rewardserver"
+	"github.com/lavanet/lava/utils"
+	conflicttypes "github.com/lavanet/lava/x/conflict/types"
+	pairingtypes "github.com/lavanet/lava/x/pairing/types"
+	spectypes "github.com/lavanet/lava/x/spec/types"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type EventTracker struct {
+	lock               sync.RWMutex
+	clientCtx          client.Context
+	blockResults       *ctypes.ResultBlockResults
+	latestUpdatedBlock int64
+}
+
+func (et *EventTracker) updateBlockResults(latestBlock int64) error {
+	ctx := context.Background()
+	blockResults, err := et.clientCtx.Client.BlockResults(ctx, &latestBlock)
+	if err != nil {
+		return err
+	}
+	// lock for update after successful block result query
+	et.lock.Lock()
+	defer et.lock.Unlock()
+	et.latestUpdatedBlock = latestBlock
+	et.blockResults = blockResults
+	return nil
+}
+
+func (et *EventTracker) getLatestPaymentEvents() (payments []*rewardserver.PaymentRequest, err error) {
+	et.lock.RLock()
+	defer et.lock.RUnlock()
+	transactionResults := et.blockResults.TxsResults
+	for _, tx := range transactionResults {
```

Comment:
what i would like to see is one loop on transaction results, have a map of event.Type to callback, and call the callback and that's it, on initialization when the updaters get the eventTracker, they call setCallbackForEvent and that's it, more efficient and cleaner class structure